### DISCUSSION TITLE
Correct way to include fipsmodule .cnf into openssl.conf Windows

### DISCUSSION THREAD

**Discussion by cassienn** [openssl/openssl#21602] [Q&A] [answered]
openssl.conf
config_diagnostics = 1
openssl_conf = openssl_init

.include "C:\\Program Files\\My application\\fipsmodule.cnf"
[openssl_init]
providers = provider_sect

[provider_sect]
fips = fips_sect
base = base_sect

[base_sect]
activate = 1

Can you please guide me what is the right way to included fipsmodule.cnf in windows environment?
1 Is the quotation mark necessary if the path has spaces?
2 Is backward slash or forward slash supported on windows environment?

Here are some examples, which of the format of absolute path is officially supported on windows environment? Or all of them are supported?
1 .include "C:\\\\Program Files\\My application\\\\fipsmodule.cnf"
2 .include C:\\\\Program Files\\\\My application\\\\fipsmodule.cnf
3 .include "C:/Program Files/My application/fipsmodule.cnf"
4 .include C:/Program Files/My application/fipsmodule.cnf
5 .include C:\Program Files\My application\fipsmodule.cnf

**Comment by t8m** [comment, marked as answer]
There should be no double quotes. Both backward and forward slashes should work as directory separator. It should not be necessary to duplicate the backward slashes although I assume if you duplicate them, it would still work.

i.e. 4 and 5 are most correct ones.

**Comment by cassienn** [comment]
There should be no double quotes. Both backward and forward slashes should work as directory separator. It should not be necessary to duplicate the backward slashes although I assume if you duplicate them, it would still work.

i.e. 4 and 5 are most correct ones.
